### PR TITLE
libc/machine: Implement ARM aeabi_xxx API called by clang

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
specified here:
https://developer.arm.com/documentation/ihi0043/latest

## Impact
new api

## Testing
Pass CI

### CODE REVIEW

File: libs/libc/machine/arm/aeabi_memset.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#include <string.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void weak_function __aeabi_memset(FAR void *s, size_t n, int c)
+{
+  memset(s, c, n);
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/aeabi_memset.c:33]: Does this implementation meets: "In general, implementations of these functions are allowed to corrupt only the integer core registers permitted to be corrupted by the [[AAPCS](https://developer.arm.com/docs/ihi0042/latest)] (r0-r3, ip, lr, and CPSR)." requirement?

File: libs/libc/machine/arm/aeabi_memclr.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * Included Files
+ ****************************************************************************/
+
+#include <string.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void weak_function __aeabi_memclr(FAR void *s, size_t n)
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/aeabi_memclr.c:31]: Does this implementation meets: "In general, implementations of these functions are allowed to corrupt only the integer core registers permitted to be corrupted by the [[AAPCS](https://developer.arm.com/docs/ihi0042/latest)] (r0-r3, ip, lr, and CPSR)." requirement?

File: libs/libc/machine/arm/aeabi_memset.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/machine/arm/aeabi_memset.c:33]: Yes, this is the standard requirement all compiler generated function and manual assemble function conform.

File: libs/libc/machine/arm/aeabi_memclr.c (hunk 2 above)
Comment (xiaoxiang781216) [reply libs/libc/machine/arm/aeabi_memclr.c:31]: ditto

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Let's optimize this to asm routines later.

**Pull request merged**